### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by simon04**
Hi @trougnouf, thanks a lot for this awesome plugin. Actually, I've long been looking for exactly this functionality. I applied various improvements to the code, please see the commit messages for details. I'd be super interested and motivated for further improvements and contributions to this plugin. Would you be interested in a contributor / co-author / co-maintainer?

### CODE REVIEW

**Comment by trougnouf** [comment]
Thank you very much for these much needed commits! I can't test at the moment because I'm bicycling around Belgium for 12 days but I looked at all the changes and it looks much better. Does it work with the vanilla multipart-post? If so it's a merge.
I would be interested :) I gave you write access to the repository.

File: mediawikiapi.lua [hunk 1]
```diff
@@ -111,7 +112,10 @@ function MediaWikiApi.uploadfile(filepath, pagetext, filename, overwrite)
     text = pagetext,
     comment = 'Uploaded with dtMediaWiki',
     token = MediaWikiApi.getEditToken(),
-    file = {filename="whatevs", file = file_handler:read("*all")},
+    file = {
+        filename = "whatevs",
+        data = file_handler:read("*all"),
```
Comment (simon04) [inline mediawikiapi.lua]: This was the only change required to make it work with vanilla multipart-post (version 1.1; tested w/ Arch Linux)

**Comment by simon04** [comment]
Thanks for your reply and your credit of trust. I'll merge my changes and might come up with some additional improvements :smile: 

I created the GitHub organisation https://github.com/dtMediaWiki and invited you as owner. You could [transfer the project there](https://help.github.com/articles/transferring-a-repository-owned-by-your-personal-account/) so we're already settled for the time this project becomes big …

**Comment by trougnouf** [comment]
Thank you :) I look forward to what you might come up with! I don't know that an organization is necessary? GitHub introduces it as "shared accounts where groups of people can collaborate across many projects at once" but I can't see this growing into multiple projects.

**Comment by simon04** [comment]
It might be overkill, you're right. Just thought I'd propose it …

In fact, I found this plugin just in time since I got big heap of climbing photos to upload … https://commons.wikimedia.org/wiki/Category:Climbing_World_Championships_2018

Enjoy your bicycle trip!

**Pull request merged**